src/teleop/gui/src/components/Amino.vue

```diff
@@ -0,0 +1,176 @@
+
+export default {
+    data() {
+        return {
+            siteIndex: 0,
+            heaters: [{
+                    enabled: false,
+                    intended: false,
+                    temp: 0,
+                    color: 'grey'
+                },
+                {
+                    enabled: false,
+                    intended: false,
+                    temp: 0,
+                    color: 'grey'
+                },
+                {
+                    enabled: false,
+                    intended: false,
+                    temp: 0,
+                    color: 'grey'
+                }
+            ],
+
+            autoShutdownEnabled: true,
+            autoShutdownIntended: true,
+
+            temp_sub: null
+        }
+    },
+
+    created: function () {
+        this.temp_sub = new ROSLIB.Topic({
+            ros: this.$ros,
+            name: 'science/temperatures',
+            messageType: 'mrover/ScienceTemperature'
+        });
+
+        this.temp_sub.subscribe((msg) => {
+            // Callback for temp_sub
+            console.log(msg)
+            for (let i = 0; i < 3; i++) {
+                this.heaters[i].temp = msg.temperatures[i]
+            }
+        });
+    },
+
+    props: {
+        site: {
+            type: String,
+            required: true,
+        }
+    },
+
+    components: {
+        ToggleButton,
+    },
+
+    methods: {
+        toggleHeater: function (id) {
+            this.heaters[id].intended = !this.heaters[id].intended
+            let toggleHeaterServ = new ROSLIB.Service({
+                ros: this.$ros,
+                name: 'change_heater_state',
+                serviceType: 'mrover/ChangeHeaterState'
+            });
+            let request = new ROSLIB.ServiceRequest({
+                device: id,
+                enable: this.heaters[id].intended
+            });
+            toggleHeaterServ.callService(request, (result) => {
+                if (!result) {
+                    alert(`Toggling heater ${id} failed.`)
+                }
+            });
+        },
+
+        sendAutoShutdownCmd: function (enabled) {
+            this.autoShutdownIntended = enabled
+            let autoShutdownServ = new ROSLIB.Service({
+                ros: this.$ros,
+                name: 'change_heater_auto_shutoff_state',
+                serviceType: 'mrover/ChangeDeviceState'
+            });
+            let request = new ROSLIB.ServiceRequest({
+                enable: this.autoShutdownIntended
+            });
+            autoShutdownServ.callService(request, (result) => {
+                if (!result) {
+                    alert(`Toggling autoshutdown failed.`)
+                }
+            });
+        }
+    },
+
+    watch: {
+        site(newVal) {
+            if (newVal === "A") {
+                this.siteIndex = 0
+            } else if (newVal === "B") {
+                this.siteIndex = 1
+            } else {
+                this.siteIndex = 2
```

Comment:
Kinda a preference-thing but I think this would make more sense as a computed variable.

It might also make more sense to do this computation in ISHTask.vue and pass both the letter and the number. Right now Amino and Sudan are doing very similar things for this and I'd rather avoid the code duplication.

I also think the mapping for this should be in the teleop yaml